### DISCUSSION TITLE
Part attributes and iBoM title

### DISCUSSION THREAD

**Discussion by Crzyrndm** [INTI-CMNB/KiBot#316] [Q&A] [answered]
Have been setting up KiBot to work with an existing project. Many thanks for the amazing tool.

I did run into a couple of niggles that I'm not sure if user error / by design / ...

* Parts/footprints in Kicad6 (5 unknown) have a set of properties listed as attributes in the GUI that look to be intended for use as a basic filter of outputs prior to variants (exclude from PCB/ Schematic / BoM / Pos / ...) so e.g. Test points can be dropped for BoM with minimal fuss. I couldn't find options to have KiBot outputs respect these?
![image](https://user-images.githubusercontent.com/8765278/195006374-d742321e-1243-4693-87f3-28bd501a73b4.png)
![image](https://user-images.githubusercontent.com/8765278/195006490-5af318c5-8bc1-40c5-aa4a-1cc18a08909a.png)
* The `ibom` output title (top left corner)is some temporary value instead of the project name if generated within KiCad. Again I was unable to find an option to control this field
![image](https://user-images.githubusercontent.com/8765278/195007356-00dc2906-5609-4ce6-9863-f597ba3f811c.png)

**Comment by set-soft** [comment, marked as answer]
Hi @Crzyrndm !

1. Nope, these attributes are ignored. Nobody mentioned using them. I'm adding some support for them.
2. This is fixed in the current git code. Note that this happens when you use some filter or variant.

**Reply by set-soft** [reply]
BTW: Why should somebody want to have an "Exclude from BoM" flag in the PCB when we have the same flag on the schematic. It looks like a source of problems.

**Reply by set-soft** [reply]
I added support for "Exclude from BOM" to the internal BoM in a3370280071e79771a8ad0ce6b6cea77795eff88
The position output already takes care of the "Exclude from position files" because this is the way KiCad 6 marks KiCad 5 *virtual* components.

**Reply by Crzyrndm** [reply]
> BTW: Why should somebody want to have an "Exclude from BoM" flag in the PCB when we have the same flag on the schematic. It looks like a source of problems.

I presume it is either needed for parts that are added directly to the PCB (not in the schematic at all), or the state is just copied from the schematic whenever PCB is updated to make the PCB more self contained. I would tend to agree that it's not how I would do it

> Hi @Crzyrndm !
> 
>     1. Nope, these attributes are ignored. Nobody mentioned using them. I'm adding some support for them.
>     2. This is fixed in the current git code. Note that this happens when you use some filter or variant.

> I added support for "Exclude from BOM" to the internal BoM in [a337028](https://github.com/INTI-CMNB/KiBot/commit/a3370280071e79771a8ad0ce6b6cea77795eff88) The position output already takes care of the "Exclude from position files" because this is the way KiCad 6 marks KiCad 5 _virtual_ components.

Thanks for the added feature/fixes and the clarification with "exclude from position files". I did kind of figure that out later while trying to figure out why certain parts made it into the pos files and others didn't even without any variant/filter (see fiducial / test point)

Thanks again for the amazing tool
